packages/components/addon/components/hds/gutters/index.js

```diff
@@ -0,0 +1,108 @@
+import Component from '@glimmer/component';
+import { htmlSafe } from '@ember/template';
+import { assert } from '@ember/debug';
+
+export const DIRECTIONS = ['horizontal', 'vertical'];
+export const SPACING_SIZES = ['xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl'];
+
+const regEx = /^-?((\d+)|(\d+\.\d+)|(\.\d+))(px|rem|em|%)$/;
+
+export default class HdsGuttersIndexComponent extends Component {
+  /**
+   * @param isSpan
+   * @type {boolean}
+   * @default false
+   * @description Indicates whether component should use a span tag instead of the default div tag.
+   */
+  get isSpan() {
+    return this.args.isSpan ?? false;
+  }
+
+  /**
+   * Get the direction for the component
+   * Accepted values: 'horizontal', 'vertical'
+   * @param direction
+   * @type {string}
+   */
+  get direction() {
+    let { direction } = this.args;
+
+    assert(
+      '@direction for "Hds::Gutters" must have a valid value',
+      DIRECTIONS.includes(direction)
+    );
+
+    return direction;
+  }
+
+  /**
+   * Get the preset spacing size to apply to the item.
+   * Accepted values: 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl'
+   * @param spacing
+   * @method presetSpacing
+   * @type {string} The class name to apply to the item.
+   * @default handled by CSS
+   */
+  get presetSpacing() {
+    let { spacing } = this.args;
```

Comment:
The spacing argument allows 2 types of values:
1. A preset spacing size. (Currently one of: "xxs", "xs", "sm", "md", "lg", "xl", "xxl")
2. A custom CSS unit. (Examples: "2.5rem", "4%", "1em", "-5px", etc.)

Depending on the type of value entered it's handled 2 different ways:
1. Preset size = appends a classname to set the spacing size
2. Custom CSS unit = Adds an inline style modifying the value of a CSS custom property.

Since "spacing" works differently depending on the type of value entered, it adds a few complications. For example: if a preset size is entered, then a class name should be added but a style attribute should not be. Conversely, if a custom CSS unit is entered, then a style attribute should be added but a class name should not be. Also if no "spacing" vale is added, then neither a class name nor a style attribute should get added to the component's underlying HTML element.

The way this is currently implemented, the different cases are taken care of. I tried to refactor to use {{style}} instead but couldn't figure out how to get it to work to handle all the various cases. I've also tried using JS .setProperty() instead to add a style attribute but I don't know how the HTML element should be targeted.

Reply:
> I tried to refactor to use {{style}} instead but couldn't figure out how to get it to work to handle all the various cases.

Use the [`style` modifier](https://github.com/jelhan/ember-style-modifier) to assign the value of `--hds-spacing` as inline style: 
```
// in .hbs file
<[div|span] class={{this.classNames}} {{style --hds-spacing=this.customSpacing}} ...attributes>
```
```
// in .js file
if (spacing !== undefined) {
  if (spacing.match(regEx)) {
    return spacing;
  }
}
```
Personally, I would also merge `presetSpacing` and `customSpacing` in a single function, that returns the spacing value (without the need for CSS classes). Of course, this also takes away the possibility of using CSS helper classes, but as mentioned elsewhere for other components we're not providing CSS classes that can be used directly as "helpers" so this would not be a bug issue (for me).

> I've also tried using JS .setProperty() instead to add a style attribute but I don't know how the HTML element should be targeted.
I've noticed that internally the [`style` modifier](https://github.com/jelhan/ember-style-modifier) is using the `setProperty()` method, so technically it would be the same as using the modifier; let's use the modifier then, which is more explicit on what it does.
https://github.com/jelhan/ember-style-modifier/blob/master/addon/modifiers/style.js#L57

Reply:
This is exactly what I tried but coudn't get to work right: `{{style --hds-spacing=this.customSpacing}}`

Anyway I'll come back to this and experiment more when I can, thanks